src/metabase/driver/mysql.clj

```diff
@@ -343,6 +347,14 @@
                                 2)
                           (hx/literal "-01"))))
 
+(defmethod sql.qp/->honeysql [:mysql :datediff]
```

Comment:
mysql is far simpler and we reuse existing function but swap the args 
`[:datediff x y unit]` -> `timestampdiff(y, x, unit)` in mysql